samcli/lib/sync/infra_sync_executor.py

```diff
@@ -0,0 +1,370 @@
+        * Location property of AWS::Serverless::Application
+        * TemplateURL property of AWS::CloudFormation::Stack
+
+        Fields skipped during template comparison because it's a metadata generated by SAM
+        * SamResourceId in Metadata property of all resources
+
+        Parameters
+        ----------
+        template_dict: Dict
+            The unprocessed template dictionary
+        linked_resources: List[str]
+            The corresponding resources in the other template that got processed
+
+        Returns
+        -------
+        List[str]
+            The list of resource IDs that got changed during sanitization
+        """
+
+        resources = template_dict.get("Resources", {})
+        processed_resources: Set[str] = set()
+
+        for resource_logical_id in resources:
+            resource_dict = resources.get(resource_logical_id, {})
+            resource_type = resource_dict.get("Type")
+
+            if resource_type in CODE_SYNCABLE_RESOURCES or resource_type in SYNCABLE_STACK_RESOURCES:
+                processed_resource = self._remove_resource_field(
+                    resource_logical_id,
+                    resource_type,
+                    resource_dict,
+                    linked_resources,
+                )
+
+                if processed_resource:
+                    LOG.debug("Sanitized %s resource %s", resource_type, resource_logical_id)
+                    processed_resources.add(processed_resource)
+
+            # Remove SamResourceId metadata since this metadata does not affect any cloud behaviour
+            resource_dict.get("Metadata", {}).pop("SamResourceId", None)
+            if not resource_dict.get("Metadata"):
+                resource_dict.pop("Metadata", None)
+            LOG.debug("Sanitizing the Metadata for resource %s", resource_logical_id)
+
+        return cast(Set, sorted(processed_resources))
+
+    def _remove_resource_field(
+        self,
+        resource_logical_id: str,
+        resource_type: str,
+        resource_dict: Dict,
+        linked_resources: Set[str] = set(),
+    ) -> Optional[str]:
+        """
+        Helper method to process resource dict
+
+        Parameters
+        ----------
+        resource_logical_id: str
+            Logical ID of the resource
+        resource_type: str
+            Resource type
+        resource_dict: Dict
+            The resource level dict containing Properties field
+        linked_resources: Set[str]
+            The corresponding resources in the other template that got processed
+
+        Returns
+        -------
+        Optional[str]
+            The processed resource ID
+        """
+        processed_logical_id = None
+
+        if resource_type == AWS_LAMBDA_FUNCTION:
+            for field in LAMBDA_FUNCTION_REMOVAL_MAP.get(resource_type, {}).get("Code", []):
+                if (
+                    is_local_path(resource_dict.get("Properties", {}).get("Code", {}).get(field, None))
```

Comment:
Somewhat related to previous comment, does `is_local_path` work as expected, if we run this block after running package context?

Reply:
Yes, it will not make sense if we use package.output_template_file anymore, I will update this section using a two template logic, using both the built template and the packaged template